File: frontend/src/types/HazardTicket.ts
export interface HazardTicket {
  id: number;
  result_id: number;
  device_id: number;
  severity: string;            // LOW / MEDIUM / HIGH / CRITICAL
  owner_id: number;
  deadline: string;
  rectify_status: string;      // OPEN / RECTIFIED / REJECTED / CLOSED
  rectify_note: string;
  rectified_at: string;
  reinspect_note: string;
  closed_at: string;
  created_at: string;
  escalated: boolean;
}

File: backend/src/models/hazard_ticket.py
from pydantic import BaseModel


class HazardTicket(BaseModel):
    id: int | float
    result_id: int | float
    severity: str
    owner_id: int | float
    deadline: str
    rectify_status: str
    rectify_note: str
    closed_at: str
    created_at: str = ""
    submitted_by: int | float | None = None
    review_note: str = ""
    inspector_id: int | float | None = None
